src/main/java/com/amazon/opendistroforelasticsearch/sql/esdomain/mapping/FieldMapping.java

```diff
@@ -0,0 +1,128 @@
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esdomain.mapping;
+
+import com.amazon.opendistroforelasticsearch.sql.domain.Field;
+import com.amazon.opendistroforelasticsearch.sql.utils.StringUtils;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.action.admin.indices.mapping.get.GetFieldMappingsResponse.FieldMappingMetaData;
+
+/**
+ * Field mapping that parses native ES mapping.
+ *
+ * NOTE that approaches in this class are NOT reliable because of the ES mapping query API used.
+ * We should deprecate this in future and parse field mapping in more solid way.
+ */
+public class FieldMapping {
+
+    /** Name of the Field to be parsed */
+    private final String fieldName;
+
+    /** Native mapping information returned from ES */
+    private final Map<String, FieldMappingMetaData> typeMappings;
+
+    /** Maps a field name to Field object that specified in query explicitly */
+    private final Map<String, Field> specifiedFieldsByName;
+
+
+    public FieldMapping(String fieldName) {
+        this(fieldName, emptyMap(), emptyMap());
+    }
+
+    public FieldMapping(String fieldName,
+                        Map<String, FieldMappingMetaData> typeMappings,
+                        Map<String, Field> specifiedFieldByNames) {
+
+        this.fieldName = fieldName;
+        this.typeMappings = typeMappings;
+        this.specifiedFieldsByName = specifiedFieldByNames;
+    }
+
+    /**
+     * Is field specified explicitly in query
+     * @return true if specified
+     */
+    public boolean isSpecified() {
+        return specifiedFieldsByName.containsKey(fieldName);
+    }
+
+    /**
+     * Verify if property field matches wildcard pattern specified in query
+     * @return true if matched
+     */
+    public boolean isWildcardSpecified() {
+        return specifiedFieldsByName.containsKey(path() + ".*");
+    }
+
+    /**
+     * Is field a property field, which means either object field or nested field.
+     * @return true for property field
+     */
+    public boolean isPropertyField() {
+        int numOfDots = StringUtils.countMatches(fieldName, '.');
+        return numOfDots > 1 || (numOfDots == 1 && !isMultiField());
+    }
+
+    /**
+     * Is field a/in multi-field, for example, field "a.keyword" in field "a"
+     * @return true for multi field
+     */
+    public boolean isMultiField() {
+        return fieldName.endsWith(".keyword");
+    }
+
+    /**
+     * Is field meta field, such as _id, _index, _source etc.
+     * @return true for meta field
+     */
+    public boolean isMetaField() {
+        return fieldName.startsWith("_");
+    }
+
+    /**
+     * Path of property field, for example "employee" in "employee.manager"
+     * @return path of property field
+     */
+    public String path() {
+        int lastDot = fieldName.lastIndexOf(".");
```

Comment:
will throw an exception when there's no dots in the fieldName.

Reply:
I did this on purpose based on the assumption that is field is property. As you're pointing out, I will add the verification. Also now I'm thinking since this method is part of the public API set, it is better to be more defensive.